### PR TITLE
Add circuit breaker

### PR DESCRIPTION
**Pull request by danpersa**

### CODE REVIEW

**Comment by NorthFury** [comment]
Shouldn't there be a different circuit breaker for different services? If the team service is down, some of the endpoints in innkeeper should still be usable.

**Comment by danpersa** [comment]
The circuit breaker is created by the provider each time it is requested.
So we have a CircuitBreaker for each of the clients.

**Comment by NorthFury** [comment]
:+1:

File: src/main/scala/org/zalando/spearheads/innkeeper/utils/UtilsModule.scala [hunk 1]
```diff
@@ -23,28 +25,48 @@ class UtilsModule extends AbstractModule with ScalaModule {
...
+}
+
+private class CircuitBreakerProvider extends Provider[CircuitBreaker] {
+
+  override def get(): CircuitBreaker = {
+    val breaker = new CircuitBreaker()
+      .withFailureThreshold(3, 5)
+      .withSuccessThreshold(3)
+      .withDelay(30L, TimeUnit.SECONDS)
+    breaker.open()
```
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/utils/UtilsModule.scala]: to overcome this: https://github.com/jhalterman/failsafe/issues/34

File: src/main/scala/org/zalando/spearheads/innkeeper/utils/UtilsModule.scala [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+import org.asynchttpclient.{DefaultAsyncHttpClient, DefaultAsyncHttpClientConfig}
 
 /**
  * @author dpersa
  */
 class UtilsModule extends AbstractModule with ScalaModule {
 
   override def configure() = {
+    bind[DefaultAsyncHttpClient].toProvider[AsyncClientProvider].asEagerSingleton()
+    bind[CircuitBreaker].toProvider[CircuitBreakerProvider]
```
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/utils/UtilsModule.scala]: this one shouldn't be singleton :)

**Comment by NorthFury** [comment]
:+1:

**Pull request merged**